cachi2/core/package_managers/npm.py

```diff
@@ -173,6 +185,193 @@ def get_sbom_components(self) -> list[dict[str, str]]:
         packages = self._dependencies if self.lockfile_version == 1 else self._packages
         return [{"name": package.name, "version": package.version} for package in packages]
 
+    def get_dependencies_to_download(self) -> Dict[str, Dict[str, Optional[str]]]:
+        """Return a Dict of URL dependencies to download."""
+        packages = self._dependencies if self.lockfile_version == 1 else self._packages
+        return {
+            package.name: {
+                "version": package.version,
+                "url": package.resolved_url,
+                "integrity": package.integrity,
+            }
+            for package in packages
+            if "file:" not in package.resolved_url
+        }
```

Comment:
Ah, this has the same issue as the `dependency_paths`

By making it a dict, you de-duplicate by name and miss dependencies

As seen in the integration tests - we download only `npm/ms-2.1.3.tgz` and `npm/ms-2.0.0.tgz` is missing

Reply:
OMG, you're right, excellent catch. Fixing and adding unit tests for it.